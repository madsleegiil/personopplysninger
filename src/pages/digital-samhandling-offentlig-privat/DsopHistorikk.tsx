import React, { useState, Fragment, useEffect } from "react";
import { DsopInfo } from "types/dsop";
import { Element } from "nav-frontend-typografi";
import moment from "moment";
import { AlertStripeInfo } from "nav-frontend-alertstriper";
import Moment from "react-moment";
import { NedChevron, OppChevron } from "nav-frontend-chevron";
import { FormattedMessage, FormattedHTMLMessage } from "react-intl";
import { Link, useLocation } from "react-router-dom";
import PageContainer from "components/pagecontainer/PageContainer";
import DSOPIkon from "assets/img/DSOP.svg";
import WithDSOP from "./DsopFetch";

interface Props {
  dsopInfo: DsopInfo;
}

/*
  Hent data
*/
const DsopHistorikk = () => (
  <PageContainer
    tittelId={"dsop.tittel"}
    icon={DSOPIkon}
    backTo={"/dsop"}
    brodsmulesti={[
      { title: "dsop.tittel", path: "/dsop" },
      { title: "dsop.levertedata" }
    ]}
  >
    <WithDSOP>{({ data }) => <Tabell dsopInfo={data} />}</WithDSOP>
  </PageContainer>
);

/*
  Visning
*/
const Tabell = (props: Props) => {
  const location = useLocation();
  const { dsopInfo } = props;

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  const initState: {
    [Ã¥r: string]: {
      dsopInnslag: DsopInfo;
      ekspandert: boolean;
    };
  } = {};

  dsopInfo.forEach((dsopInnslag, i) => {
    const year = moment(dsopInnslag.uthentingsTidspunkt).year();

    if (!initState[year]) {
      initState[year] = {
        dsopInnslag: [dsopInnslag],
        ekspandert: !i
      };
    } else {
      initState[year].dsopInnslag.push(dsopInnslag);
    }
  });

  const [data, setData] = useState(initState);

  return (
    <div className="historikk__tabs-innhold historikk__flex-table">
      {Object.keys(data).length > 0 ? (
        <>
          <div className={"historikk__info"}>
            <AlertStripeInfo>
              <FormattedHTMLMessage id={"eksternelenker.dsop.info"} />
            </AlertStripeInfo>
          </div>
          <div className="historikk__flex-rad historikk__head">
            <div className="historikk__flex-kolonne">
              <Element>
                <FormattedMessage id="dsop.uthentingstidspunkt" />
              </Element>
            </div>
            <div className="historikk__flex-kolonne">
              <Element>
                <FormattedMessage id="dsop.mottaker" />
              </Element>
            </div>
          </div>
          {Object.keys(data)
            .reverse()
            .map(year => {
              const value = data[year];
              const onClick = () =>
                setData({
                  ...data,
                  [year]: {
                    ...data[year],
                    ekspandert: !data[year].ekspandert
                  }
                });

              return (
                <Fragment key={year}>
                  <div className="historikk__flex-rad" key={year}>
                    <div
                      className="historikk__flex-kolonne af-liste__ekspander"
                      onClick={onClick}
                    >
                      {year}{" "}
                      {value.ekspandert ? <OppChevron /> : <NedChevron />}
                    </div>
                    <div />
                  </div>
                  {value.ekspandert &&
                    value.dsopInnslag.map((dsopInnslag, i) => (
                      <div className="historikk__flex-rad" key={i}>
                        <div className="historikk__flex-kolonne historikk__heading">
                          <Moment format="DD.MM - hh:mm">
                            {dsopInnslag.uthentingsTidspunkt}
                          </Moment>
                        </div>
                        <div className="historikk__flex-kolonne">
                          <Link
                            to={`${location.pathname}/${dsopInnslag.uthentingsTidspunkt}`}
                            className="lenke"
                          >
                            {dsopInnslag.mottakernavn}
                          </Link>
                        </div>
                      </div>
                    ))}
                </Fragment>
              );
            })}
        </>
      ) : (
        <div className="historikk__ingen-data">
          <AlertStripeInfo>
            <FormattedMessage id="dsop.ingendata" />
          </AlertStripeInfo>
        </div>
      )}
    </div>
  );
};

export default DsopHistorikk;
