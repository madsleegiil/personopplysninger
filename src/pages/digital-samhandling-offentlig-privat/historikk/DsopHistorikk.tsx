import React from "react";
import PageContainer from "components/pagecontainer/PageContainer";
import DSOPIkon from "assets/img/DSOP.svg";
import WithDSOP from "../DsopFetch";
import DsopHistorikkView from "./DsopHistorikkView";

const DsopHistorikk = () => (
  <PageContainer
    tittelId={"dsop.tittel"}
    icon={DSOPIkon}
    backTo={"/#flere-opplysninger"}
    brodsmulesti={[{ title: "dsop.tittel" }]}
  >
    <WithDSOP>{({ data }) => <DsopHistorikkView dsopInfo={data} />}</WithDSOP>
  </PageContainer>
);
export default DsopHistorikk;
