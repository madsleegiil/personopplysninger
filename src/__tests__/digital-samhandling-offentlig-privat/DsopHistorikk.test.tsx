import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import dsopInfo from "../../clients/apiMock/app/fetch/dsop-info.json";
import { IntlProvider } from "react-intl";
import nbMessages from "text/nb";
import DsopHistorikk from "../../pages/digital-samhandling-offentlig-privat/historikk/DsopHistorikk";

jest.mock("react-router-dom", () => ({
  Link: () => <div />,
  useLocation: () => ({
    pathname: "pathname",
  }),
}));

describe("DsopHistorikk", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <IntlProvider locale={"nb"} messages={nbMessages}>
          <DsopHistorikk dsopInfo={dsopInfo} />
        </IntlProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
