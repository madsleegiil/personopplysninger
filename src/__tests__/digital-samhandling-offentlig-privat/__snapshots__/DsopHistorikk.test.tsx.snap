// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DsopHistorikkView renders correctly 1`] = `
<div
  className="historikk__tabs-innhold historikk__flex-table"
>
  <div
    className="historikk__info"
  >
    <div
      className="alertstripe alertstripe--info"
    >
      <span
        className="alertstripe__ikon"
      >
        <span
          className="sr-only"
        >
          info
        </span>
        <svg
          aria-label="info-ikon"
          focusable="false"
          height="1.5em"
          kind="info-sirkel-fyll"
          role="img"
          viewBox="0 0 24 24"
          width="1.5em"
        >
          <g
            fill="none"
          >
            <path
              d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
              fill="#337C9B"
            />
            <path
              d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
              fill="#FFF"
            />
          </g>
        </svg>
      </span>
      <div
        className="typo-normal alertstripe__tekst"
      >
        Dette er en logg som viser hvilke opplysninger vi har utlevert til forsikringsselskap basert på samtykke du har gitt på Altinn. 
        <a
          className="lenke"
          href="https://altinn.no"
          rel="noopener"
          target="blank"
        >
          Se samtykket i Altinn.
        </a>
      </div>
    </div>
  </div>
  <div
    className="historikk__flex-rad historikk__head"
  >
    <div
      className="historikk__flex-kolonne"
    >
      <p
        className="typo-element"
      >
        Uthentingstidspunkt
      </p>
    </div>
    <div
      className="historikk__flex-kolonne"
    >
      <p
        className="typo-element"
      >
        Selskap
      </p>
    </div>
  </div>
  <div
    className="historikk__flex-rad"
  >
    <div
      className="historikk__flex-kolonne af-liste__ekspander"
      onClick={[Function]}
    >
      2019
       
      <span
        className="nav-frontend-chevron chevronboks chevron--opp"
      />
    </div>
    <div />
  </div>
  <div
    className="historikk__flex-rad"
  >
    <div
      className="historikk__flex-kolonne historikk__heading"
    >
      <time
        dateTime={"2019-06-18T08:45:51.634Z"}
      >
        18.06 - 10:45
      </time>
    </div>
    <div
      className="historikk__flex-kolonne"
    >
      <div />
    </div>
  </div>
  <div
    className="historikk__flex-rad"
  >
    <div
      className="historikk__flex-kolonne historikk__heading"
    >
      <time
        dateTime={"2019-06-18T08:45:52.101Z"}
      >
        18.06 - 10:45
      </time>
    </div>
    <div
      className="historikk__flex-kolonne"
    >
      <div />
    </div>
  </div>
  <div
    className="historikk__flex-rad"
  >
    <div
      className="historikk__flex-kolonne historikk__heading"
    >
      <time
        dateTime={"2019-06-18T08:51:55.535Z"}
      >
        18.06 - 10:51
      </time>
    </div>
    <div
      className="historikk__flex-kolonne"
    >
      <div />
    </div>
  </div>
  <div
    className="historikk__flex-rad"
  >
    <div
      className="historikk__flex-kolonne af-liste__ekspander"
      onClick={[Function]}
    >
      2018
       
      <span
        className="nav-frontend-chevron chevronboks chevron--ned"
      />
    </div>
    <div />
  </div>
  <div
    className="historikk__flex-rad"
  >
    <div
      className="historikk__flex-kolonne af-liste__ekspander"
      onClick={[Function]}
    >
      2017
       
      <span
        className="nav-frontend-chevron chevronboks chevron--ned"
      />
    </div>
    <div />
  </div>
  <div
    className="historikk__flex-rad"
  >
    <div
      className="historikk__flex-kolonne af-liste__ekspander"
      onClick={[Function]}
    >
      2016
       
      <span
        className="nav-frontend-chevron chevronboks chevron--ned"
      />
    </div>
    <div />
  </div>
</div>
`;
